plugins/by-name/lspsaga/deprecations.nix

```diff
@@ -0,0 +1,205 @@
+lib: {
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "lspsaga"
+      ];
+    in
+    map
+      (
+        optionName:
+        # See https://github.com/NixOS/nixpkgs/pull/247916
+        # TODO remove those warnings in ~2 months (October 2023)
+        lib.mkRemovedOptionModule (basePluginPath ++ [ optionName ]) ''
+          The `lspsaga` plugin used by default in Nixvim has changed to https://github.com/nvimdev/lspsaga.nvim.
+          Please, adapt your config to the new options.
+          Documentation available at https://nix-community.github.io/nixvim/plugins/lspsaga/index.html.
+        ''
+      )
+      [
+        "signs"
+        "headers"
+        "maxDialogWidth"
+        "icons"
+        "maxFinderPreviewLines"
+        "keys"
+        "borderStyle"
+        "renamePromptPrefix"
+      ]
```

Comment:
I know our deprecation policy discussion is still open, however this was added August 2023 and has been in 23.11, 24.05, 24.11, and 25.05.

I don't think any maintainers would argue for keeping it at this point 🙂